src/syslogd.c

```diff
@@ -3271,6 +3301,58 @@ static void logit(char *fmt, ...)
 	fflush(stdout);
 }
 
+static void notifier_add(struct notifiers *newn, const char *program)
+{
+	while (*program && isspace(*program))
+		program++;
+
+	/* Check whether it is accessible, regardless of TOCTOU */
+	if (!access(program, X_OK)) {
+		struct notifier *np;
+		size_t len;
+
+		len = strlen(program);
+
+		np = calloc(1, sizeof(*np) + len +1);
+		if (np) {
+			/* xxx Actually wastes space -- vararray? */
+			np->n_program = (char*)&np[1];
+			memcpy(np->n_program, program, len);
+			SIMPLEQ_INSERT_TAIL(newn, np, n_link);
+		} else
+			ERR("Cannot allocate memory for a notify program");
+	} else
+		logit("notify: non-existing, or not executable program\n");
+}
+
+static void notifier_invoke(const char *logfile)
+{
+	char *argv[3];
+	struct notifier *np;
+
+	logit("notify: rotated %s, invoking hooks\n", logfile);
+
+	SIMPLEQ_FOREACH(np, &nothead, n_link) {
+		switch (fork()) {
+		case -1:
+			ERR("Cannot start notifier %s", np->n_program);
+			break;
+		case 0:
+			/* No specific I/O setup, just use what we had */
+			argv[0] = np->n_program;
+			argv[1] = (char*)logfile; /* logical unconst */
+			argv[2] = NULL;
+			execv(argv[0], argv);
+			_exit(1);
+		default:
+			/* Do not care beside that, no special process group
+			 * etc.; it will eventually be reaped via reapchild() */
```

Comment:
It's perfectly possible to reap children that are in their own process group.  I do that, e.g., in Finit.  So the comment is either redundant or incomplete.  I suggest dropping this comment too.